### PR TITLE
Add a sticky footer

### PR DESCRIPTION
**Pull request by ItzLukaDev**
Added a sticky footer. Approved by Fisher on the gurkult discord.

### CODE REVIEW

**Review by Arnav-2004 — CHANGES_REQUESTED** [review]
Haven't gone through full PR but here are few changes that I noticed.

File: reagurk/src/components/footer.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+export {};
```
Comment (Arnav-2004) [inline reagurk/src/components/footer.tsx:3]: I don't think this is required.

```suggestion
```

File: reagurk/src/components/footer.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+export {};
+
+export default function Footer(): JSX.Element {
+
+  return (
+
+    <>
+    
+      <footer className="text-gray-400 bg-gray-800 body-font fixed
```
Comment (Arnav-2004) [inline reagurk/src/components/footer.tsx:11]: Format properly.

```suggestion
    <>
      <footer className="text-gray-400 bg-gray-800 body-font fixed
```

File: reagurk/src/components/footer.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+               
+            </a>
+            
+          </span>
+        </div>
+      </footer> 
+    </>
+
+  ); 
+
```
Comment (Arnav-2004) [inline reagurk/src/components/footer.tsx:39]: Format properly.

```suggestion
  ); 
```

**Review by Arnav-2004 — CHANGES_REQUESTED** [review]

**Comment by JustARandomDeveloper** [comment]
Cool the PR done now?

**Review by fisher60 — CHANGES_REQUESTED** [review]
This needs work for mobile responsiveness. On my 720 monitor, the footer collapses onto the hero banner as shown with my attachment.

Mobile also allows the banner to collapse on itself, we probably need to make sure the page can still scroll if the banner + footer would be overlapping otherwise

720p screen
image

Mobile (iphone x)
image

File: reagurk/src/components/footer.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+          <p className="text-sm text-gray-400 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-800 sm:py-2 sm:mt-0 mt-4">© 2021 Gurkult 
+           
+          </p>
+          <span className="inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start">
+            <a className="text-gray-400"
+              href='https://github.com/gurkult'>
+              <img src="https://img.icons8.com/ios-glyphs/50/ffffff/github.png" alt="github logo" />
+            </a>
+            <a className="ml-3 text-gray-400"
+              href='https://discord.gg/W9DSfryp8Y'>
```
Comment (fisher60) [inline reagurk/src/components/footer.tsx:25]: Sorry, missed this one. This should reference the constant we have defined for the server invite.

File: reagurk/src/components/footer.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+export default function Footer(): JSX.Element {
+
+  return (
+    <>
+      <footer className="text-gray-400 bg-gray-800 body-font fixed
+             inset-x-0
+             bottom-0">
+        <div className="container px-5 py-8 mx-auto flex items-center sm:flex-row flex-col">
+          <a className="flex title-font font-medium items-center md:justify-start justify-center text-white">
+            <img src="https://cdn.discordapp.com/emojis/792081164466126868.png?size=56" alt="gurkult logo" />
```
Comment (fisher60) [inline reagurk/src/components/footer.tsx:13]: The discord CDN should not be considered reliable as image hosting for us. If there is an image we need to use, we should have it locally.

File: reagurk/src/components/footer.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            <span className="ml-3 text-xl">Gurkult</span>
+          </a>
+          <p className="text-sm text-gray-400 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-800 sm:py-2 sm:mt-0 mt-4">© 2021 Gurkult 
+           
+          </p>
+          <span className="inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start">
+            <a className="text-gray-400"
+              href='https://github.com/gurkult'>
```
Comment (fisher60) [inline reagurk/src/components/footer.tsx:21]: This should use a constant

File: reagurk/src/components/footer.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+              <img src="https://img.icons8.com/ios-glyphs/50/ffffff/github.png" alt="github logo" />
+            </a>
+            <a className="ml-3 text-gray-400"
+              href='https://discord.gg/W9DSfryp8Y'>
+              <img src="https://img.icons8.com/fluency-systems-filled/48/ffffff/discord-logo.png"/>
```
Comment (fisher60) [inline reagurk/src/components/footer.tsx:26]: Maybe worth downloading

**Review by fisher60 — CHANGES_REQUESTED** [review]
Sorry, missed a few things and went through the code again with new review comments.

**Comment by dawnofmidnight** [comment]
note: These images are not accurate, and we intend to use the SVGs, I just need to get around to normalizing them (ie. using the `fill` attribute as opposed to css`)/making sure they work right.

**Comment by PandaTheProgrammer** [comment]
So what now?

File: reagurk/src/components/footer.tsx (hunk 6 above)
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:21]: What? I don't see which constant are you talking about.

File: reagurk/src/components/footer.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            <img src="https://cdn.discordapp.com/emojis/792081164466126868.png?size=56" alt="gurkult logo" />
+            <span className="ml-3 text-xl">Gurkult</span>
+          </a>
+          <p className="text-sm text-gray-400 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-800 sm:py-2 sm:mt-0 mt-4">© 2021 Gurkult 
+           
+          </p>
+          <span className="inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start">
+            <a className="text-gray-400"
+              href='https://github.com/gurkult'>
+              <img src="https://img.icons8.com/ios-glyphs/50/ffffff/github.png" alt="github logo" />
```
[the remark this answers is not in the document]
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:22]: Ok

File: reagurk/src/components/footer.tsx (hunk 4 above)
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:25]: Ah

File: reagurk/src/pages/App.tsx [hunk 9]
```diff
@@ -1,11 +1,13 @@
...
+import Footer from '../components/footer';
 
 export default function App(): JSX.Element {
   return (
     <div className="App">
       <body className="relative bg-white dark:bg-gray-800 overflow-hidden">
         <Hero />
+        <Footer/>
       </body>
     </div>
```
[the remark this answers is not in the document]
Comment (PandaTheProgrammer) [reply reagurk/src/pages/App.tsx]: Ok..done.

**Comment by PandaTheProgrammer** [comment]
> I don't get why its like that on a 720p screen let me see mine

**Comment by PandaTheProgrammer** [comment]
image

**Comment by PandaTheProgrammer** [comment]
image

**Comment by PandaTheProgrammer** [comment]
What's the issue with the mobile?

File: reagurk/src/components/footer.tsx (hunk 6 above)
Comment (dawnofmidnight) [reply reagurk/src/components/footer.tsx:21]: > What? I don't see which constant are you talking about.

https://github.com/gurkult/reagurk/blob/bc256c14d1ff300d316f7b9b8ec70ba3e468fd97/reagurk/src/constants.ts#L1

https://github.com/gurkult/reagurk/blob/bc256c14d1ff300d316f7b9b8ec70ba3e468fd97/reagurk/src/constants.ts#L3

File: reagurk/src/components/footer.tsx (hunk 5 above)
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:13]: Resolved.

File: reagurk/src/components/footer.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:22]: Fixed.

**Comment by PandaTheProgrammer** [comment]
> note: These images are not accurate, and we intend to use the SVGs, I just need to get around to normalizing them (ie. using the `fill` attribute as opposed to css`)/making sure they work right.

Fixed.

File: reagurk/src/components/footer.tsx (hunk 7 above)
Comment (PandaTheProgrammer) [reply reagurk/src/components/footer.tsx:26]: Fixed.

**Comment by PandaTheProgrammer** [comment]
Any more changes?

**Review by fisher60 — CHANGES_REQUESTED** [review]
I am still seeing some responsiveness changes with mobile and smaller screen sizes where the footer takes up too much screen space.

This should be fixed by allowing the footer to scroll if the screen would be too small to comfortably accommodate it. I will provide examples in the Discord channel.

File: reagurk/src/components/footer.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+
+
+export default function Footer(): JSX.Element {
+
+  return (
+
+    <>
+    
+      <footer className="text-gray-400 bg-gray-800 body-font fixed
```
Comment (fisher60) [inline reagurk/src/components/footer.tsx:22]:
```suggestion
      <footer className="text-gray-400 bg-gray-800 body-font md:fixed
```

**Review by fisher60 — CHANGES_REQUESTED** [review]
Sorry for the extended review here, I am a bit to blame as I have not had a time to get a good look at this PR or the tailwind framework. Please look into the changes I suggested as this will allow scrolling on longer/narrow screens. This does not fix the 720p issue I outlined, so it might worth getting this perfect, but I am not overly concerned with supporting this niche screen size.

File: reagurk/src/components/footer.tsx (hunk 10 above)
Comment (OldManHacker) [reply reagurk/src/components/footer.tsx]: I added that. Works!

**Comment by OldManHacker** [comment]
image


for me, 720 resolution works fine.

**Comment by OldManHacker** [comment]
> 


this is 1280*720?

**Review by fisher60 — APPROVED** [review]
This looks good to me. Thank you and awesome job, really appreciate to working with us through the reviews.

**Pull request merged**